### PR TITLE
Use bytes for signature instead of v,r,s

### PR DESCRIPTION
**Pull request by kevincheng96**
`QuarkWallets` currently take `v`, `r`, `s` as inputs for the ECDSA signature. This limits the wallet to only support ECDSA signatures, even if the `signer` of the wallet is a EIP-1271 smart contract that supports other signature schemes (e.g. BLS, Passkey). To make the wallet more flexible and be able to support other signature schemes, we change the signature input to a generic `bytes` type.

Note: This only increases flexibility for Quark wallets that have an EIP-1271 contract as a signer, where the EIP-1271 contract is upgradeable. Quark wallets owned by EOA signers will still need to be migrated over to a new wallet implementation in order to support other types of signature schemes. 

Gas impact: Seems like this change bumps up gas usage by 900-1k (negligible). This is likely because the wallet now needs to take the extra step of decoding `bytes signature` into `v,r,s`. The tests that fail the gas snapshot are all ones that run `executeQuarkOperation` multiple times, exceeding the gas diff tolerance.

### CODE REVIEW

File: src/quark-core/src/QuarkWallet.sol [hunk 1]
```diff
@@ -448,7 +409,8 @@ contract QuarkWallet is IERC1271 {
                 revert InvalidEIP1271Signature();
             }
         } else {
-            (address recoveredSigner, ECDSA.RecoverError recoverError) = ECDSA.tryRecover(digest, v, r, s);
+            // For EOA signers, this implementation of the QuarkWallet only supports ECDSA signatures
+            (address recoveredSigner, ECDSA.RecoverError recoverError) = ECDSA.tryRecover(digest, signature);
```
Comment (fluffywaffles) [inline src/quark-core/src/QuarkWallet.sol:413]: oh nice, OZ did the decoding for us :+1:

**Review by fluffywaffles — APPROVED** [review]
LGTM

File: src/quark-core/src/QuarkWallet.sol [hunk 2]
```diff
@@ -171,58 +171,48 @@ contract QuarkWallet is IERC1271 {
...
 
     /**
      * @notice Executes a first play or a replay of a QuarkOperation via signature
      * @dev Can only be called with signatures from the wallet's signer
      * @param op A QuarkOperation struct
      * @param submissionToken The submission token for the replayable quark operation for QuarkNonceManager. This is initially the `op.nonce`, and for replayable operations, it is the next token in the nonce chain.
-     * @param v EIP-712 signature v value
-     * @param r EIP-712 signature r value
-     * @param s EIP-712 signature s value
+     * @param signature A EIP-712 signature
```
Comment (hayesgm) [inline src/quark-core/src/QuarkWallet.sol:189]: A digital siganture, e.g. EIP-712?

**Review by hayesgm — APPROVED** [review]
looks good. one comment on the natspec

File: src/quark-core/src/QuarkWallet.sol (hunk 2 above)
Comment (kevincheng96) [reply src/quark-core/src/QuarkWallet.sol]: Good call. These no longer have to be EIP-712 sigs. Updated.

**Pull request merged**